prow/pjutil/filter.go

```diff
@@ -51,19 +52,25 @@ func AvailablePresubmits(changes config.ChangedFilesProvider, org, repo, branch
 	}
 	runWithTrigger, err := FilterPresubmits(AggregateFilter(triggerFilters), changes, branch, presubmits, logger)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	var runWithTestAllNames []string
+	var runWithTestAllCommands []string
 	for _, ps := range runWithTestAll {
-		runWithTestAllNames = append(runWithTestAllNames, ps.Name)
+		runWithTestAllCommands = append(runWithTestAllCommands, ps.Name)
 	}
-	var runWithTriggerNames []string
+	var optionalJobsWithTriggerCommands []string
+	var requiredJobsWithTriggerCommands []string
 	for _, ps := range runWithTrigger {
-		runWithTriggerNames = append(runWithTriggerNames, ps.RerunCommand)
+		if ps.Optional {
+			optionalJobsWithTriggerCommands = append(optionalJobsWithTriggerCommands, ps.RerunCommand)
+		} else {
+			requiredJobsWithTriggerCommands = append(requiredJobsWithTriggerCommands, ps.RerunCommand)
+		}
+
 	}
 
-	return runWithTestAllNames, runWithTriggerNames, nil
+	return runWithTestAllCommands, optionalJobsWithTriggerCommands, requiredJobsWithTriggerCommands, nil
```

Comment:
naming is hard nits:
- `runWithTestAllCommands` doesn't make sense to me as a name here, it's going to be a list of `["jobA", "jobB"]`... I think `runWithTestAllNames` still makes sense here
- `optionalJobsWithTriggerCommands` seems like it should be `optionalJobTriggerCommands` since it will be a list of `["/test jobA", "/test jobB"]`
- ditto for required jobs